### PR TITLE
First pass at Connect Events CLI

### PR DESCRIPTION
**Pull request by lambacck**

### CODE REVIEW

File: lib/connect/api.js [hunk 1]
```diff
@@ -76,6 +87,53 @@ let withMapping = exports.withMapping = function (connection, objectName) {
...
+    let stream
+    connection.streams.forEach(function (s) {
+      if (s.object_name.toLowerCase().indexOf(objectNameLower) === 0) {
+        stream = s
+      }
+    })
+    if (stream !== undefined) {
+      return yield Promise.resolve(stream)
+    } else {
+      throw new Error('No stream configured for ' + objectName)
```
Comment (mattrothenberg) [inline lib/connect/api.js]: Style tweak. I think we should use string interpolation where possible –

```js
throw new Error(`No stream configured for ${objectName}`)
```

File: commands/connect-events/resume.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  flags: [
+    {name: 'resource', description: 'specific connection resource name', hasValue: true}
+  ],
+  needsApp: true,
+  needsAuth: true,
+  run: cli.command(co.wrap(function * (context, heroku) {
+    cli.action('resuming connection', co(function * () {
+      let connection = yield api.withConnection(context, heroku, api.ADDON_TYPE_EVENTS)
+      context.region = connection.region_url
+      let url = '/api/v3/kafka-connections/' + connection.id + '/actions/resume'
```
Comment (mattrothenberg) [inline commands/connect-events/resume.js]: Let's use string interpolation here, too.

```js
let url = `/api/v3/kafka-connections/${connection.id}/actions/resume`
```

File: commands/connect-events/recover.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  flags: [
+    {name: 'resource', description: 'specific connection resource name', hasValue: true}
+  ],
+  needsApp: true,
+  needsAuth: true,
+  run: cli.command(co.wrap(function * (context, heroku) {
+    yield cli.action('recovering connection', co(function * () {
+      let connection = yield api.withConnection(context, heroku)
+      context.region = connection.region_url
+      let url = '/api/v3/connections/' + connection.id + '/actions/restart'
```
Comment (mattrothenberg) [inline commands/connect-events/recover.js]: String interpolation.

```js
let url = `/api/v3/connections/${connection.id}/actions/restart`
```

File: commands/connect-events/pause.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  flags: [
+    {name: 'resource', description: 'specific connection resource name', hasValue: true}
+  ],
+  needsApp: true,
+  needsAuth: true,
+  run: cli.command(co.wrap(function * (context, heroku) {
+    yield cli.action('pausing connection', co(function * () {
+      let connection = yield api.withConnection(context, heroku, api.ADDON_TYPE_EVENTS)
+      context.region = connection.region_url
+      let url = '/api/v3/kafka-connections/' + connection.id + '/actions/pause'
```
Comment (mattrothenberg) [inline commands/connect-events/pause.js]: String interpolation

File: commands/connect-events/db-set.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        default: context.flags.schema || 'salesforce',
+        when: !context.flags.schema
+      }
+    ]).then(co.wrap(function * (answers) {
+      for (let key in answers) {
+        data[key] = answers[key]
+      }
+
+      yield cli.action('setting database parameters', co(function * () {
+        let url = '/api/v3/kafka-connections/' + connection.id
```
Comment (mattrothenberg) [inline commands/connect-events/db-set.js]: String interpolation

**Comment by mattrothenberg** [comment]
This might be my lack of familiarity with the Heroku CLI, but I wonder if `heroku connect-events:streams --help` should list out what the available commands are? It's unclear, currently.

Screen Shot 2019-06-07 at 9 55 21 AM

**Comment by lambacck** [comment]
I don't disagree with all the style changes you requested, but all those instances are straight up copies from the equivalent connect:* commands.

**Comment by mattrothenberg** [comment]
@lambacck Fair point! Don't let this be a blocker. I'm happy with either choice.

**Comment by lambacck** [comment]
@mattrothenber maybe I messed something up about the stream sub command. This works:
```
heroku connect-events:stream --help
Add a stream

USAGE
  $ heroku connect-events:stream:COMMAND

DESCRIPTION
  Add a stream

COMMANDS
  connect-events:stream:add     Add a stream
  connect-events:stream:delete  Delete an existing stream
  connect-events:stream:state   return a stream state
```

**Comment by lambacck** [comment]
I think I figured it out. I had streams instead of stream in the topic naming in index.js.

**Pull request merged**